src/canmatrix/canmatrix.py

```diff
@@ -552,45 +584,54 @@ def globSignals(self, globStr):
 
     def addAttribute(self, attribute, value):
         """
-        add attribute to attribute-list of frame; If Attribute already exits, modify value
+        Add the attribute with value to customer Frame attribute-list. If Attribute already exits, modify its value.
+        :param str attribute: Attribute name
+        :param any value: attribute value
         """
         try:
             self.attributes[attribute] = str(value)
-        except:
+        except:  # could it really fail?
```

Comment:
I'm not sure why I added this try-catch. But I'm quite sure that there was a problem without it.
Maybe there is a problem with "str" if your are in python2 and have some issues with charsets...

Reply:
None of this should be dealt with in this PR but:

1) it should catch the particular exception not just anything
2) i think users should be expected to pass the proper type
3) i think there are still various str-or-int-or-... around so maybe it would get caught up in cleaning that up.